packages/calcite-components/src/components/block-group/block-group.e2e.ts

```diff
@@ -0,0 +1,550 @@
+        const blockGroups = document.querySelectorAll("calcite-block-group");
+        blockGroups.forEach((blockGroup) =>
+          blockGroup.addEventListener("calciteBlockGroupOrderChange", () => {
+            testWindow.calledTimes++;
+          }),
+        );
+      });
+
+      await dragAndDrop(
+        page,
+        {
+          element: `calcite-block[heading="d"]`,
+          shadow: "calcite-sort-handle",
+        },
+        {
+          element: `#first-letters`,
+          pointerPosition: {
+            vertical: "bottom",
+          },
+        },
+      );
+
+      await dragAndDrop(
+        page,
+        {
+          element: `calcite-block[heading="e"]`,
+          shadow: "calcite-sort-handle",
+        },
+        {
+          element: `#numbers`,
+          pointerPosition: {
+            vertical: "bottom",
+          },
+        },
+      );
+
+      await dragAndDrop(
+        page,
+        {
+          element: `calcite-block[heading="e"]`,
+          shadow: "calcite-sort-handle",
+        },
+        {
+          element: `#no-group`,
+          pointerPosition: {
+            vertical: "bottom",
+          },
+        },
+      );
+
+      const [first, second, third, fourth, fifth, sixth, seventh, eight, ninth] = await page.findAll("calcite-block");
+      expect(await first.getProperty("heading")).toBe("a");
+      expect(await second.getProperty("heading")).toBe("b");
+      expect(await third.getProperty("heading")).toBe("d");
+      expect(await fourth.getProperty("heading")).toBe("1");
+      expect(await fifth.getProperty("heading")).toBe("2");
+      expect(await sixth.getProperty("heading")).toBe("no-group");
+      expect(await seventh.getProperty("heading")).toBe("c");
+      expect(await eight.getProperty("heading")).toBe("e");
+      expect(await ninth.getProperty("heading")).toBe("f");
+
+      expect(await page.evaluate(() => (window as TestWindow).calledTimes)).toBe(2);
+    });
+
+    it("reorders using a keyboard", async () => {
+      const page = await createSimpleBlockGroup();
+
+      let totalMoves = 0;
+
+      // Workaround for page.spyOnEvent() failing due to drag event payload being serialized and there being circular JSON structures from the payload elements. See: https://github.com/Esri/calcite-design-system/issues/7643
+      await page.$eval("calcite-block-group", (blockGroup: BlockGroup["el"]) => {
+        const testWindow = window as TestWindow;
+        testWindow.calledTimes = 0;
+        blockGroup.addEventListener("calciteBlockGroupOrderChange", (event: CustomEvent<BlockDragDetail>) => {
+          testWindow.calledTimes++;
+          testWindow.newIndex = event.detail.newIndex;
+          testWindow.oldIndex = event.detail.oldIndex;
+          testWindow.fromEl = event.detail.fromEl.id;
+          testWindow.toEl = event.detail.toEl.id;
+          testWindow.el = event.detail.dragEl.id;
+        });
+      });
+
+      async function assertReorder(
```

Comment:
Now that list and block-group assert on reorder and move, let's add some test utils. Can you create an issue for this?

Reply:
https://github.com/Esri/calcite-design-system/issues/11435